### PR TITLE
CB-20158 Handle parcel download errors in poller

### PR DESCRIPTION
**Pull request by Bajzathd**
See detailed description in the commit message.

### CODE REVIEW

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerUpgradeParcelDownloadListenerTask.java [hunk 1]
```diff
@@ -37,14 +48,15 @@ public ClouderaManagerUpgradeParcelDownloadListenerTask(ClouderaManagerApiPojoFa
...
-        String parcelStage = apiParcel.getStage();
-
-        if (!ParcelStatus.DOWNLOADED.name().equals(parcelStage)
-                && !ParcelStatus.DISTRIBUTED.name().equals(parcelStage)
-                && !ParcelStatus.DISTRIBUTING.name().equals(parcelStage)
-                && !ParcelStatus.ACTIVATED.name().equals(parcelStage)
-                && !ParcelStatus.ACTIVATING.name().equals(parcelStage)) {
-            LOGGER.warn("Expected parcel status is {}, received status is: {}", ParcelStatus.DOWNLOADED.name(), parcelStage);
+
+        List<String> errors = apiParcel.getState().getErrors();
```
Comment (pdarvasi) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerUpgradeParcelDownloadListenerTask.java:52]: this logic should be included in `ClouderaManagerUpgradeParcelDistributeListenerTask` and `ClouderaManagerSingleParcelActivationListenerTask`, too, could you pls do that or would you prefer a follow-up PR?

File: cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        lenient().when(clouderaManagerApiPojoFactory.getParcelResourceApi(apiClient)).thenReturn(parcelResourceApi);
+        lenient().when(parcelResourceApi.readParcel(anyString(), anyString(), anyString())).thenReturn(apiParcel);
+        lenient().when(apiParcel.getState()).thenReturn(apiParcelState);
+        lenient().when(apiParcelState.getErrors()).thenReturn(List.of());
+    }
+
+    @Test
+    void shouldHandleParcelStateAndStageNull() {
+        when(apiParcel.getStage()).thenReturn(null);
+        when(apiParcel.getState()).thenReturn(null);
```
Comment (pdarvasi) [inline cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java:78]: duplicate line

File: cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    ApiParcelState apiParcelState;
+
+    abstract T setUpUnderTest();
+
+    @BeforeEach
+    void setUp() throws Exception {
+        underTest = setUpUnderTest();
+        lenient().when(pollerObject.getApiClient()).thenReturn(apiClient);
+        lenient().when(clouderaManagerApiPojoFactory.getParcelResourceApi(apiClient)).thenReturn(parcelResourceApi);
+        lenient().when(parcelResourceApi.readParcel(anyString(), anyString(), anyString())).thenReturn(apiParcel);
```
Comment (pdarvasi) [inline cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java:70]: minor: `lenient()`-s are not needed here

File: cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerSingleParcelActivationListenerTaskTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+class ClouderaManagerSingleParcelActivationListenerTaskTest
+        extends AbstractClouderaManagerParcelListenerTaskTest<ClouderaManagerSingleParcelActivationListenerTask> {
+
+    @Mock
+    private StackDto stackDto;
+
+    @Override
+    ClouderaManagerSingleParcelActivationListenerTask setUpUnderTest() {
+        lenient().when(stackDto.getName()).thenReturn(CLUSTER_NAME);
+        lenient().when(pollerObject.getStack()).thenReturn(stackDto);
```
Comment (pdarvasi) [inline cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/ClouderaManagerSingleParcelActivationListenerTaskTest.java:19]: `lenient()`-s are not needed here

File: cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java (hunk 2 above)
Comment (Bajzathd) [reply cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java:78]: not really, **getStage** and **getState**

File: cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java (hunk 3 above)
Comment (Bajzathd) [reply cluster-cm/src/test/java/com/sequenceiq/cloudbreak/cm/polling/task/AbstractClouderaManagerParcelListenerTaskTest.java:70]: fixed, thanks

**Review by pdarvasi — APPROVED** [review]
Nicely done, thanks for the generic solution!

**Pull request merged**